cmd/object-handlers.go

```diff
@@ -2506,30 +2521,64 @@ func (api objectAPIHandlers) PutObjectRetentionHandler(w http.ResponseWriter, r
 		writeErrorResponse(ctx, w, errorCodes.ToAPIErr(ErrServerNotInitialized), r.URL, guessIsBrowserReq(r))
 		return
 	}
+	// Get Content-Md5 sent by client and verify if valid
```

Comment:
We should perform request validation after confirming that the requester has the permission to perform PutObjectRetention request.

Reply:
done